### PR TITLE
[tune] add-note command for Tune CLI

### PR DESCRIPTION
**Pull request by andrewztan**
## What do these changes do?

Adds `add-note` command to Tune CLI. Builds on top of #3983

### CODE REVIEW

**Comment by andrewztan** [comment]
@richardliaw

**Comment by andrewztan** [comment]
@richardliaw 
Questions:
1. Should we limit to just txt files?
2. Should the --filename flag be the full filename (e.g. note.txt) or just name (e.g. note), or allow for both?
3. In addition to experiment folders, should add-note be allowed in project or other folders?
4. What feedback do we want to give the user during and after the note editing experience?

**Comment by richardliaw** [comment]
> Questions:
> 1. Should we limit to just txt files?
Well, the default should just be `note.txt` for now. I don't think you need to worry about otherwise. 

> 2. Should the --filename flag be the full filename (e.g. note.txt) or just name (e.g. note), or allow for both?

Full name should be good.

> 3. In addition to experiment folders, should add-note be allowed in project or other folders?

Yeah, that should be fine. 

> 4. What feedback do we want to give the user during and after the note editing experience?
What do you have in mind for this?

**Comment by hartikainen** [comment]
This is nice! I think it would also be great to be able to add notes when launching an experiment.

File: python/ray/tune/scripts.py [hunk 1]
```diff
@@ -29,10 +29,23 @@ def list_experiments(project_path, sort):
     commands.list_experiments(project_path, sort)
 
 
+@cli.command()
+@click.argument("path", required=True, type=str)
+@click.option(
+    '--filename',
```
Comment (richardliaw) [inline python/ray/tune/scripts.py]:
```suggestion
    "--filename",
```

**Review by richardliaw — APPROVED** [review]
I think this looks fine for now; I'll try this out tomorrow.

I think later on this should maybe somehow use a starter template?

**Comment by richardliaw** [comment]
> This is nice! I think it would also be great to be able to add notes when launching an experiment.

Hm, via `run_experiments` as a config like `note=str`?

**Comment by hartikainen** [comment]
@richardliaw Yeah, something like `note=str` would be nice, although I'm not sure if adding things like that might make the api cluttered at some point.

**Comment by richardliaw** [comment]
@hartikainen I'll merge this for now (just as a preliminary first cut to get something in), but in a later PR we'll expose the contents of this note file through `tune ls` and then we can add a Python API for modifying after iterating.

**Comment by richardliaw** [comment]
@andrewztan nice work!

**Pull request merged**